### ISSUE TITLE
STaR-GATE

### ISSUE THREAD

**Issue by JennyHuang19** [JennyHuang19/counterfactual_feedback#40]
teaching models to ask the correct clarifying questions.

0. create ground truth labels: the ideal response given prompts and personas.
1. prompt the llm to ask good questions.
2. filter for conversations that contain "good" questions (this means high P(gold response | prompt, persona)).
3. generate the model response conditioned on the best conversations.
4. fine-tune on (conversations (including good clarifying questions, responses)).

**Comment by JennyHuang19** [comment]
goal: we would like to probe users in a way that is max info gain and efficiency.

- an influx of clarification questions may annoy the user to the point of being counterproductive.

**Comment by JennyHuang19** [comment]
what triggers a clarifying question, and how do we know it is the right time for a particular question? 

although the paper prompts an LM to ask good questions, much of the work seems to be done in the prompt. and then we are optimizing for (filtering for) getting the ground-truth answer.

regularization: stop the model after 3 turns. examine its response. this provides an explicit boundary on the number of question-answer pairs generated in the training data.

**Comment by JennyHuang19** [comment]
problem:
dependence on gold responses.

solution:
RLHI with user-based rewards.
1. user personas: from a long conversation history, summarize the user's preferences  (e.g., "Favors answers with numbers, statistics, and concrete evidence"). summary seems to focus on the style of the response.
- dimensions: tone, structure, informativeness
2. explicitly condition the reward on the user persona.
4. use the user-conditioned reward model to score the responses (sample N, the preference pair is the lowest and highest-scoring responses).

**Comment by JennyHuang19** [comment]
We observe that: 
(1) Some users provide little feedback, while others reveal clear and consistent behaviors; (2) Many personas reflect common expectations, yet a notable subset exhibit unique preferences (e.g., repeatedly requesting analogies or engaging in role-play with recurring characters); and 
(3) Some users’ needs vary across domains
